src/toxcore/dht/server/mod.rs

```diff
@@ -187,10 +189,10 @@ impl Server {
             onion_symmetric_key: Arc::new(RwLock::new(secretbox::gen_key())),
             onion_announce: Arc::new(RwLock::new(OnionAnnounce::new(pk))),
             friends: Arc::new(RwLock::new(friends)),
-            nodes_to_bootstrap: Arc::new(RwLock::new(Bucket::new(None))),
+            nodes_to_bootstrap: Arc::new(RwLock::new(Bucket::new(MAX_TO_BOOTSTRAP))),
```

Comment:
Why do we keep them in a Bucket instead of HashSet?

Reply:
We use try_add() of Bucket.

Reply:
Then we don’t have to use try_bucket. What's the point?

Reply:
Because we can't afford to store all nodes - it can lead to memory overflow. So we have to limit the size somehow. c-toxcore leaves only the closest nodes, so do we. But I'm not sure if it's correct and how it's related to the eclipse attack. Anyway it's the topic for another PR - I don't like that we use for these lists full DhtNode either so I'll try to change it somehow (though not sure yet how).